src/main/webapp/PlaceGuideOnList.js

```diff
@@ -0,0 +1,41 @@
+class PlaceGuideOnList {
+  constructor(placeGuideId, place, name, creator, description, audioKey, audioLength, imageKey, createdByCurrentUser) {
+    const placeGuideDiv = document.createElement("div");
+    placeGuideDiv.setAttribute('id', placeGuideId);
+    
+    placeGuideDiv.addEventListener('click', function() {
+      PlaceGuideManager.highlightPlaceGuide(placeGuideId);
+    });
+
+    createBlobView(audioKey, "audio", placeGuideDiv);
+    createBlobView(imageKey, "img", placeGuideDiv);
+
+    if (createdByCurrentUser) {
+      const deleteButton = document.createElement("button");
+      deleteButton.addEventListener("click", function() {
+        if (window.confirm("Click ok if you want to delete the place guide")) {
+          PlaceGuideManager.remove(placeGuideId);
+        }
+      });
+
+      const editButton = document.createElement("button");
+      const queryString = 
+      editButton.addEventListener("click", function() {
+        window.location = "./createPlaceGuide.html?";
+      });
```

Comment:
This one should also be a separate function. The constructor would be much easier to follow if you wouldn't have so many different variables, listeners, messages, etc, just a few function calls with suggestive names for what functionality a particular function is adding.